test/tests/functional/pbs_job_array_comment.py

```diff
@@ -0,0 +1,94 @@
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+class TestJobComment(TestFunctional):
+    """
+    Testing job array comment is accurate
+    """
+
+    def test_job_array_comment(self):
+        """
+        Testing job array comment is correct when one or more sub jobs
+        are rejected by mom
+        """
+        attr = {'resources_available.ncpus': 10}
+        self.server.manager(MGR_CMD_SET, NODE, attr)
+        attr = {'job_history_enable': 'true'}
+        self.server.manager(MGR_CMD_SET, SERVER, attr)
+        # create a mom hook that rejects and deletes subjob 0, 5, and 7
+        hook_name = "reject_subjob"
+        hook_body = (
+            "import pbs\n"
+            "import re\n"
+            "e = pbs.event()\n"
+            "jid = str(e.job.id)\n"
+            "if re.match(r'[0-9]*\[[057]\]', jid):\n"
+            "    e.job.delete()\n"
+            "    e.reject()\n"
+            "else:\n"
+            "    e.accept()\n"
+        )
+        attr = {'event': 'execjob_begin', 'enabled': 'True'}
+        self.server.create_import_hook(hook_name, attr, hook_body)
+
+        test_job_array = Job(TEST_USER, attrs={
+            ATTR_J: '0-9',
+            'Resource_List.select': 'ncpus=1'
+        })
+        test_job_array.set_sleep_time(1)
+        jid = self.server.submit(test_job_array)
+        attr = {
+            ATTR_state: 'B',
+            ATTR_comment: (MATCH_RE, 'Job Array Began at .*')
+        }
+        self.server.expect(JOB, attr, id=jid, attrop=PTL_AND)
+        self.server.expect(JOB, {'comment': 'Subjob failed'}, offset=1,
+                           id=create_subjob_jid(jid, 0), extend='x')
+        self.server.expect(JOB, {'comment': 'Subjob failed'}, offset=1,
+                           id=create_subjob_jid(jid, 5), extend='x')
+        self.server.expect(JOB, {'comment': 'Subjob failed'}, offset=1,
+                           id=create_subjob_jid(jid, 7), extend='x')
```

Comment:
Say ATTR_comment for the above three expect() calls

Reply:
Fixed.